### DISCUSSION TITLE
how can I access nuxt.config.js from component?

### DISCUSSION THREAD

**Discussion by tasiotas** [nuxt/nuxt#17401] [Questions] [answered]
Hi,

I have a NavBar.vue component, where I would like to set colorMode propety defined in defineNuxtConfig.

**nuxt.config.ts**
```js
export default defineNuxtConfig({
  inkline: {
    css: true,
    components: {},
    colorMode: 'light',
  },
```

How can I access config from component?
I thought I could use `const nuxtApp = useNuxtApp()`, but that throws error:

![image](https://user-images.githubusercontent.com/9046931/174665288-168fe025-01df-4c7f-bd97-b051e865fdf7.png)

**Comment by danielroe** [comment, marked as answer]
You won't be able to access your Nuxt config within your app, by design.

What information are you trying to access, and why?

**Reply by tasiotas** [reply]
This guide shows how to toggle dark `colorMode` in Vue3/Inkline.
I am trying to apply same logic in Nuxt, but obviously I took a wrong approach.

https://www.inkline.io/docs/introduction/guides/dark-mode

Since `colorMode` is defined in nuxt.config.ts, how else can I set its value from app ?

**Reply by danielroe** [reply]
You should do it exactly as in their guide.

**Reply by tasiotas** [reply]
Thank you! 
I have never seen or used `inject<Prototype>`. After a closer look and fresh morning read of docs I got it working.
